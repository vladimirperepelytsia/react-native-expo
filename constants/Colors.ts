export const Colors = {
  light: {
    text: '#11181C',
    background: '#fff',
    baseBorderColor: '#e3e3e3',
    green: '#40BF80',
    tint: '#222',
    icon: '#687076',
    tabIconDefault: '#687076',
    tabIconSelected: '#222',
    Switch: {
      trackColor: { true: '#006ae6', false: '#e3e3e3' },
      thumbColor: '#fff',
    },
  },
  dark: {
    text: '#ECEDEE',
    background: '#151718',
    baseBorderColor: '#e3e3e3',
    green: '#40BF80',
    tint: '#fff',
    icon: '#9BA1A6',
    tabIconDefault: '#9BA1A6',
    tabIconSelected: '#fff',
    Switch: {
      trackColor: { true: '#006ae6', false: '#fff' },
      thumbColor: '#fff',
    },
  },
};

export default Colors;
