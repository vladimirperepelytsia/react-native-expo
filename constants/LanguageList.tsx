const LanguageList = [
  {
    id: 1,
    title: 'English',
    value: 'en',
  },
  {
    id: 2,
    title: 'Українська',
    value: 'ua',
  },
];

export default LanguageList;
