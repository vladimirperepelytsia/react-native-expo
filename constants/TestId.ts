export const testIdHomePage = 'homePage';
export const testIdSettingPage = 'SettingPage';
export const testIdButtonBack = 'ButtonBack';
