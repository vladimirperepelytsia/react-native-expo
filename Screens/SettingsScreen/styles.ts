import { StyleSheet } from 'react-native';

import type { ColorSchemeName } from 'react-native/Libraries/Utilities/Appearance';

import Colors from '@/constants/Colors';

const getStyles = ({ theme }:{ theme?: ColorSchemeName }) => StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors[theme ?? 'light'].background,
    padding: 12,
    gap: 24,
  },
  switchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 12,
    backgroundColor: Colors[theme ?? 'light'].background,
  },
});

export default getStyles;
