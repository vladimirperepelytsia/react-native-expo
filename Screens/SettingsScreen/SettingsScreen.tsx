import { useEffect, useState } from 'react';
import { Appearance, View } from 'react-native';

import getStyles from './styles';

import AppSwitch from '@/components/AppSwitch/AppSwitch';
import AppText from '@/components/AppText/AppText';
import setColorScheme = Appearance.setColorScheme;
import { useTranslation } from 'react-i18next';

import LanguageSelector from '@/components/LanguageSelector/LanguageSelector';
import { testIdSettingPage } from '@/constants/TestId';
import { useAppDispatch, useAppSelector } from '@/store/hooks/useApp';
import selectCurrentTheme from '@/store/slices/theme/selectors';
import { toggleTheme } from '@/store/slices/theme/slice';

export default function SettingsScreen() {
  const theme = useAppSelector(selectCurrentTheme);
  const dispatch = useAppDispatch();
  const { t } = useTranslation();
  const [isEnabled, setIsEnabled] = useState(false);
  const styles = getStyles({ theme });

  useEffect(() => {
    setColorScheme?.(isEnabled ? 'dark' : 'light');
  }, [isEnabled]);

  return (
    <View style={styles.container} testID={testIdSettingPage}>
      <View style={styles.switchContainer}>
        <AppSwitch
          toggleSwitch={(prop) => {
            setIsEnabled(prop);
            dispatch(toggleTheme());
          }}
          isEnabled={theme === 'dark'}
        />
        <AppText>{t('settings.darkMode')}</AppText>
      </View>
      <LanguageSelector />
    </View>
  );
}
