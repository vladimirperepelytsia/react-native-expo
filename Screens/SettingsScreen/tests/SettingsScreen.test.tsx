import SettingsScreen from '../SettingsScreen';

import { render } from '@/__tests__/test-utils';
import { testIdSettingPage } from '@/constants/TestId';

describe('<SettingsScreen />', () => {
  test('should render component SettingsScreen', () => {
    const tree = render(<SettingsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render component SettingsScreen with text', () => {
    const { getByTestId } = render(<SettingsScreen />);
    getByTestId(testIdSettingPage);
  });
});
