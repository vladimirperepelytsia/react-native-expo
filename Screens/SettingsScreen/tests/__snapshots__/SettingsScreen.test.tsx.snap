// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> should render component SettingsScreen 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "gap": 24,
      "padding": 12,
    }
  }
  testID="SettingPage"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "gap": 12,
      }
    }
  >
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#006ae6"
      style={
        {
          "height": 31,
          "width": 51,
        }
      }
      thumbTintColor="#fff"
      tintColor="#e3e3e3"
      value={false}
    />
    <Text
      style={
        [
          {
            "color": "#11181C",
            "fontFamily": "UbuntuRegular",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Dark mode
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "gap": 24,
      }
    }
    data={
      [
        {
          "id": 1,
          "title": "English",
          "value": "en",
        },
        {
          "id": 2,
          "title": "Українська",
          "value": "ua",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flexDirection": "row",
              "gap": 24,
              "opacity": 1,
            }
          }
          testID="language-selector-en"
        >
          <View
            style={
              {
                "backgroundColor": "#40BF80",
                "borderColor": "#e3e3e3",
                "borderRadius": 10,
                "borderWidth": 3,
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              [
                {
                  "color": "#11181C",
                  "fontFamily": "UbuntuRegular",
                  "fontSize": 16,
                },
                undefined,
              ]
            }
          >
            English
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flexDirection": "row",
              "gap": 24,
              "opacity": 1,
            }
          }
          testID="language-selector-ua"
        >
          <View
            style={
              {
                "borderColor": "#e3e3e3",
                "borderRadius": 10,
                "borderWidth": 3,
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              [
                {
                  "color": "#11181C",
                  "fontFamily": "UbuntuRegular",
                  "fontSize": 16,
                },
                undefined,
              ]
            }
          >
            Українська
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
